src/client/client.cpp

```diff
@@ -5,22 +5,84 @@
 #include "passive_conn.hpp"
 #include "request.hpp"
 #include "text_data.hpp"
+#include "logger.hpp"
+#include "fail_if.hpp"
+#include "job_bus.hpp"
 
+#include <iostream>
 #include <vector>
 
 /* Public */
 
 Client::Client(int t_port) : server_conn(t_port, new TextIO()){
-   auto ip = std::string("127.0.0.1");
+   
+   if ( t_port == 0 ) {
+      return;
+   }
+   
+   std::string response = "";
+   LOG_INFO("Connecting to server...");
+
+   auto ip = std::string("206.189.0.154");
    req = server_conn.connect_socket(ip);
+   
+   server_conn.receive(req);
+   
+   FAIL_IF( false == req.m_valid);
+   
+   response = TextData::to_string(req.data());
+
+   FAIL_IF_MSG( false == valid_response(Reply::r_200, response),"Client didn't connect to the server");
+
+   LOG_INFO("%s! Client connected to server succesfully", response.c_str());
+
+   return;
+
+fail:
+   exit(1);
 }
 
-bool Client::ping() { return true; }
+Client::~Client(){
+   LOG_INFO("Disconnecting from server");
+   
+   // JobBus::set_exit();
+   
+   std::string command = "EXIT";
+   req.set_data(new TextData(command));
+
+   server_conn.respond(req);
+   server_conn.receive(req);
 
+   std::string response = TextData::to_string(req.data());
+   LOG_INFO("Server reply => %s", response.c_str());
+
+   close(server_conn.get_socket());
+
+   LOG_INFO("Client disconnected\n");
+}
+
+bool Client::ping() { 
+  req.set_data(new TextData("PING"));
+   
+   server_conn.respond(req);
+   server_conn.receive(req);
+   
+   std::cout << "server reply => " << TextData::to_string(req.data());
+   
+   return true; 
+}
 
 /* Contact direct */
-std::vector<std::string> Client::contact_get_contacts() const {
-   return m_contacts.get_contacts();
+std::vector<std::string> Client::contact_get_contacts() {
+   return m_contacts.display_contacts();
+}
+
+std::string Client::contact_get_current_contact() {
+   return m_contacts.get_current_contact();
+}
+
+void Client::contact_set_current_contact(std::string &t_current_contact) {
+   m_contacts.set_current_contact(t_current_contact);
```

Comment:
In my opinion you are creating a many layers of [indirection](https://tinyurl.com/3deu9jp4) without a clear reason.

` JobBus::contact_get_contacts() > Client::contact_get_contacts() >  Contacts::get_current_contact() `

Seems like `Client` only job is to delegate the functionality to `Contact`.    

Or maybe `Client` methods could be static and populate `JobBus::m_JobBus_map` so you wouldn't need the `JobBus` static methods.